src/libs/actions/App.js

```diff
@@ -85,36 +98,48 @@ AppState.addEventListener('change', (nextAppState) => {
 
 /**
  * Fetches data needed for app initialization
- *
- * @param {Boolean} shouldSyncPolicyList Should be false if the initial policy needs to be created. Otherwise, should be true.
  * @returns {Promise}
  */
-function getAppData(shouldSyncPolicyList = true) {
+function getAppData() {
     BankAccounts.fetchUserWallet();
 
-    if (shouldSyncPolicyList) {
-        Policy.getPolicyList();
-    }
-
     // We should update the syncing indicator when personal details and reports are both done fetching.
     return Promise.all([
         PersonalDetails.fetchPersonalDetails(),
         Report.fetchAllReports(true, true),
     ]);
 }
 
+/**
+ * Gets a comma separated list of locally stored policy ids
+ *
+ * @param {Array} policies
+ * @return {String}
+ */
+function getPolicyIDList(policies) {
+    return _.chain(policies)
+        .filter(Boolean)
+        .map(policy => policy.id)
+        .join(',');
+}
+
 /**
  * Fetches data needed for app initialization
+ * @param {Array} policies
  */
-function openApp() {
-    API.read('OpenApp');
+function openApp(policies) {
+    API.read('OpenApp', {
+        policyIDList: getPolicyIDList(policies),
```

Comment:
why doesn't this read `allPolicies` like `reconnectApp` does?

Reply:
We use `withOnyx` because that guarantess the following flow: read data from local storage -> render component -> trigger API call. So the `policyIDList` is always loaded

`Onyx.connect` does not guarantee that the data is available and the call could be made without the policyIDList. Unfortunately I couldn't find a different way to use withOnyx in `ReconnectApp` since it's not called from within the component.

Reply:
@aldo-expensify let me know if that doesn't make sense and I'll try to clarify some more.

Reply:
It does make sense, thanks for explaining :)

Reply:
Resolving this thread. Can you give it another review then?